kroxylicious/src/main/java/io/kroxylicious/proxy/future/BrutalFuture.java

```diff
@@ -0,0 +1,31 @@
+package io.kroxylicious.proxy.future;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+public class BrutalFuture<T> extends CompletableFuture<T> {
+
+    public boolean internalComplete(T value) {
+        return super.complete(value);
+    }
+
+    public boolean internalCompleteExceptionally(Throwable ex) {
+        return super.completeExceptionally(ex);
+    }
+
+    @Override
+    public boolean complete(T value) {
+        throw new RuntimeException("don't complete me");
```

Comment:
Did you consider just returning `false` rather than throwing an exception?  FWIW I think throwing the exception is better because we are interested in preventing our user from calling the methods at all.  I wanted to voice the question.

Reply:
No, I hadn't thought of it. I think it's better to fail explosively to make it visible to the user that we are changing some core contracts of CompletableFuture (that it can be completed).

I'm also adding debug logging to the future so you should be able to see if spewing with the log levels up.